lib/ipam/ipam.go

```diff
@@ -601,12 +607,6 @@ func (c ipamClient) AssignIP(ctx context.Context, args AssignIPArgs) error {
 			}
 
 			log.Debugf("Block for IP %s does not yet exist, creating", args.IP)
-			cfg, err := c.GetIPAMConfig(ctx)
```

Comment:
thanks. allocationBlock.assign method used the block.StrictAffinity (which we removed) and I replaced it with IPAM.StrictAffinity so I moved out of the block to have it available in case of no error. I will add it back also inside the loop to address your comment.